cachi2/core/package_managers/yarn/project.py

```diff
@@ -126,3 +143,54 @@ def yarn_cache(self) -> RootedPath:
     def from_source_dir(cls, source_dir: RootedPath) -> "Project":
         """Create a Project from a sources directory path."""
         return cls(source_dir, NotImplemented, NotImplemented)
+
+
+def get_semver_from_yarn_path(yarn_path: Optional[str]) -> Optional[semver.version.Version]:
+    """Parse yarnPath from yarnrc and return a semver Version if possible else None."""
+    if not yarn_path:
+        return None
+
+    # https://github.com/yarnpkg/berry/blob/2dc59443e541098bc0104d97b5fc452781c64baf/packages/plugin-essentials/sources/commands/set/version.ts#L208
+    yarn_spec_pattern = re.compile(r"^yarn-(.+)\.cjs$")
+    match = yarn_spec_pattern.match(Path(yarn_path).name)
+    if not match:
+        return None
+
+    yarn_version = match.group(1)
+    try:
+        return semver.version.Version.parse(yarn_version)
+    except ValueError:
+        return None
```

Comment:
Yeah, I agree. Some reaction/notification seems appropriate here. I suppose I lean towards a warning rather than a fail because even though we can't parse yarnPath, packageManager may still be usable.

Does that sound good?

Reply:
The concern is that the yarn version specified in `packageManager` may behave differently than the one in `yarnPath`. But yeah, a warning is probably enough

Reply:
Added :+1: